src/lib/models/outoforder/Core.cc

```diff
@@ -263,9 +259,31 @@ void Core::raiseException(const std::shared_ptr<Instruction>& instruction) {
 }
 
 void Core::handleException() {
+  // Check the buffer entries to see if they are branch instructions.  If so,
+  // flush them from the BP
+  for (size_t slot = 0; slot < fetchToDecodeBuffer_.getWidth(); slot++) {
+    auto& macroOp = fetchToDecodeBuffer_.getTailSlots()[slot];
+    if (!macroOp.empty() && macroOp[0]->isBranch()) {
+      predictor_.flush(macroOp[0]->getInstructionAddress());
+    }
+    macroOp = fetchToDecodeBuffer_.getHeadSlots()[slot];
+    if (!macroOp.empty() && macroOp[0]->isBranch()) {
+      predictor_.flush(macroOp[0]->getInstructionAddress());
+    }
+  }
   fetchToDecodeBuffer_.fill({});
```

Comment:
It's not entirely generic because some contain macroops and some microops.  However, I've added two functions to PipelineBuffer.hh for each of these cases to cleanup the Core.cc file a little.

Reply:
handleException looks much cleaner now